frontend/public/components/secrets/create-secret/OpaqueSecretFormEntry.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { DroppableFileInput } from './DropableFileInput';
+import { OpaqueSecretFormEntryProps } from './types';
+import { Base64 } from 'js-base64';
+
+export const OpaqueSecretFormEntry: React.FC<OpaqueSecretFormEntryProps> = ({
+  onChange,
+  entry,
+  index,
+}) => {
+  const { t } = useTranslation();
+
+  const handleValueChange = (fileData: string, isBinary: boolean) => {
+    const updatedEntry = {
+      ...entry,
+      value: isBinary ? Base64.decode(fileData) : fileData,
```

Comment:
Here is the logic of handling binary file uploads I mentioned in the previous review response, this basically decodes the contents of the binary file and stores them in this form into `opaqueDataEntries` for a brief moment (until component rerenders and the entries are set using the `base64Data` prop). I'm not sure if this is the right way to go at this, I debated myself if it wouldn't be better approach to store all the values in base64 encoding and pretty much do an inverse logic.

Reply:
Tried to reflect on this and I added some suggestions to the changes here, that are functioning fine locally for me. Was this what you meant by not decoding binary data and filling data fields with "base64StringData" property?

Reply:
Tried to reflect on this and I added some suggestions to the changes here, that are functioning fine locally for me. Was this what you meant by not decoding binary data and filling data fields with "base64StringData" property?